packages/terra-functional-testing/tests/jest/services/wdio-docker-service.test.js

```diff
@@ -0,0 +1,302 @@
+      jest.spyOn(service, 'deployStack').mockImplementationOnce(() => Promise.resolve());
+
+      await service.onPrepare({});
+
+      expect(service.deployStack).toHaveBeenCalled();
+    });
+  });
+
+  describe('initializeSwarm', () => {
+    it('should initialize the docker swarm if not already active', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve({ stdout: JSON.stringify({ Swarm: { LocalNodeState: 'inactive' } }) }));
+
+      await service.initializeSwarm();
+
+      expect(mockExec).toHaveBeenCalledWith('docker info --format "{{json .}}"');
+      expect(mockExec).toHaveBeenCalledWith('docker swarm init');
+    });
+
+    it('should not initialize the docker swarm if already active', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve({ stdout: JSON.stringify({ Swarm: { LocalNodeState: 'active' } }) }));
+
+      await service.initializeSwarm();
+
+      expect(mockExec).toHaveBeenCalledWith('docker info --format "{{json .}}"');
+      expect(mockExec).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('deployStack', () => {
+    it('should deploy the stack', async () => {
+      const service = new DockerService();
+
+      jest.spyOn(service, 'removeStack').mockImplementationOnce(() => Promise.resolve());
+      jest.spyOn(service, 'awaitNetworkReady').mockImplementationOnce(() => Promise.resolve());
+
+      await service.deployStack();
+
+      const composeFilePath = path.resolve(__dirname, '../../../lib/docker/docker-compose.yml');
+
+      expect(mockExec).toHaveBeenCalledWith(`docker stack deploy -c ${composeFilePath} wdio`);
+    });
+
+    it('should remove the previous network stack', async () => {
+      const service = new DockerService();
+
+      jest.spyOn(service, 'removeStack').mockImplementationOnce(() => Promise.resolve());
+      jest.spyOn(service, 'awaitNetworkReady').mockImplementationOnce(() => Promise.resolve());
+
+      await service.deployStack();
+
+      expect(service.removeStack).toHaveBeenCalled();
+    });
+
+    it('should wait for the network to become ready', async () => {
+      const service = new DockerService();
+
+      jest.spyOn(service, 'removeStack').mockImplementationOnce(() => Promise.resolve());
+      jest.spyOn(service, 'awaitNetworkReady').mockImplementationOnce(() => Promise.resolve());
+
+      await service.deployStack();
+
+      expect(service.awaitNetworkReady).toHaveBeenCalled();
+    });
+  });
+
+  describe('removeStack', () => {
+    it('should not execute the docker stack removal if it is already removed', async () => {
+      const service = new DockerService();
+
+      mockExec.mockImplementation(() => Promise.resolve({ stdout: '' }));
+
+      await service.removeStack();
+
+      expect(mockExec).toHaveBeenCalledTimes(1);
+      expect(mockExec).toHaveBeenCalledWith('docker stack ls | grep wdio || true');
```

Comment:
```suggestion
      expect(mockExec).toHaveBeenCalledWith('docker stack ls | grep wdio || true');
       expect(mockExec).not.toHaveBeenCalledWith('docker stack rm wdio');
```

Reply:
Added the additional check:

https://github.com/cerner/terra-toolkit/pull/412/commits/81ec6b8b522dd42261437c12f789f6e373615c9a